packages/mongo/mongo_driver.js

```diff
@@ -151,13 +151,10 @@ MongoConnection = function (url, options) {
     mongoOptions.db.native_parser = false;
   }
 
-  // XXX maybe we should have a better way of allowing users to configure the
-  // underlying Mongo driver
+  // Internally the oplog connections specify their own poolSize
+  // which we don't want to overwrite with any user defined value
   if (_.has(options, 'poolSize')) {
-    // If we just set this for "server", replSet will override it. If we just
-    // set it for replSet, it will be ignored if we're not using a replSet.
-    mongoOptions.server.poolSize = options.poolSize;
-    mongoOptions.replSet.poolSize = options.poolSize;
+    mongoOptions.poolSize = options.poolSize;
```

Comment:
Oh right yeah, well as far as I've looked into things it seems like legacy options will work just fine, though I'm not totally sure what happens with `safe`, I think it may just be ignored as it's the default on later Mongo, but I could be wrong.